### PR TITLE
Add separate content filter class to Civil Tongue Ex

### PR DESCRIPTION
**Pull request by igraziatto**
This PR is a dependency for: https://github.com/vanillaforums/collegeconfidential/pull/362

**Steps to test:**
- In the [collegeconfidential](https://github.com/vanillaforums/collegeconfidential) repo checkout `fix/hobsons-civiltongueex` if it's not merged yet.
- Enable [Hobsons](https://github.com/vanillaforums/collegeconfidential/tree/master/plugins/Hobsons) plugin
- Enable CivilTongueEX
- Add some works in the blacklist and a replacement
- Make sure you have a discussion with some bad words in the title.
- Verify if the replacement displays properly and in the comment header section (between the comment and the discussion):
Screen Shot 2020-03-04 at 12 07 12 PM

**Further testing:**
- Disable CivilTongueEX and make sure Hobsons doesn't crash for disabling a dependency.

### CODE REVIEW

File: plugins/CivilTongueEx/class.civiltongueex.plugin.php [hunk 1]
```diff
@@ -214,8 +221,8 @@ public function discussionsController_render_before($sender) {
         $discussions = $sender->data('Discussions', []);
         if (is_array($discussions) || $discussions instanceof \Traversable) {
             foreach ($discussions as &$discussion) {
-                $discussion->Name = $this->replace($discussion->Name);
-                $discussion->Body = $this->replace($discussion->Body);
+                $discussion->Name = $this->contentFilter->replace($discussion->Name);
```
Comment (dannimajdalani) [inline plugins/CivilTongueEx/class.civiltongueex.plugin.php]: We are calling $this->replace function that calls contentFilter all over the plugin, can't we do the same here?

Comment (igraziatto) [reply plugins/CivilTongueEx/class.civiltongueex.plugin.php]: Sure.

File: plugins/CivilTongueEx/class.civiltongueex.plugin.php [hunk 2]
```diff
@@ -15,12 +17,17 @@ class CivilTonguePlugin extends Gdn_Plugin {
     /** @var mixed  */
     public $Replacement;
 
+    /** @var ContentFilterInterface */
+    protected $contentFilter;
+
     /**
-     *
+     * CivilTonguePlugin constructor.
```
Comment (dannimajdalani) [inline plugins/CivilTongueEx/class.civiltongueex.plugin.php]: Might need a new line after CivilTonguePlugin constructor.

File: plugins/CivilTongueEx/Library/ContentFilter.php [hunk 3]
```diff
@@ -42,7 +42,7 @@ public function replace($text = '') {
      *
      * @return array
      */
-    public function getPatterns() {
+    public function getPatterns():array {
```
Comment (dannimajdalani) [inline plugins/CivilTongueEx/Library/ContentFilter.php]: public function getPatterns(): array {

File: plugins/CivilTongueEx/ContentFilterInterface.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Plugins;
+
+/**
+ * Interface ContentFilterInterface
+ * @package Vanilla\Plugins\CivilTonguePlugin
+ */
+interface ContentFilterInterface {
```
[the remark this answers is not in the document]
Comment (igraziatto) [reply plugins/CivilTongueEx/ContentFilterInterface.php]: You can use outside the addon if the addon is enabled.

**Pull request merged**